Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5630,4 +5648,14 @@ std::optional<LayoutUnit> RenderBox::explicitIntrinsicInnerHeight() const
     return std::optional<LayoutUnit> { height->value() };
 }
 
+RenderBlockFlow* RenderBox::blockFormattingContextRoot() const
+{
+    // This method should probably not be called on the initial containing block
+    ASSERT(!is<HTMLHtmlElement>(element()));
+    auto cb = containingBlock();
+    while (cb && !cb->createsBlockFormattingContext())
+        cb = cb->containingBlock();
+    return dynamicDowncast<RenderBlockFlow>(cb);
+}
```

Comment:
I can't find any caller of this function.

Reply:
Should be used now with some other changes